internal/controller/ols_app_redis_assets.go

```diff
@@ -0,0 +1,102 @@
+package controller
+
+import (
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *OLSConfigReconciler) generateRedisDeployment(cr *olsv1alpha1.OLSConfig) (*appsv1.Deployment, error) {
+	cacheReplicas := int32(1)
+	DeploymentSelectorLabels := map[string]string{
+		"app.kubernetes.io/component":  "redis-server",
+		"app.kubernetes.io/managed-by": "lightspeed-operator",
+		"app.kubernetes.io/name":       "lightspeed-service-redis",
+		"app.kubernetes.io/part-of":    "openshift-lightspeed",
+	}
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OLSAppRedisDeploymentName,
+			Namespace: cr.Namespace,
+			Labels:    DeploymentSelectorLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &cacheReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: DeploymentSelectorLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: DeploymentSelectorLabels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
```

Comment:
Do we need pass the configurations in `RedisCacheConfig` to this Redis server container? So that the settings are synchronized on both application side and cache side.

Reply:
IMHO, No. User should make sure that their `RedisCacheConfig` is inline with their redis settings. We cannot modify everything from redis standpoint based on user configuration input, which requires changes at multiple places in redis configuration which is not possible. (i.e. for example redis port)

Reply:
We may need an option to tell the operator not to deploy a Redis server, for the case that the user use another Redis server than the one the operator deploys.

Reply:
@raptorsun I like that option , but I dont think we planned that part of this story. what do you think ?

Reply:
Taking it in a different story sounds good to me. We already have decent amount of changes here.

Reply:
Checked just now. This upcoming story seems close to this usecase: https://issues.redhat.com/browse/OLS-221, which is on my next action items list.